### ISSUE TITLE
Build our own docker images for integration testing in CI

### ISSUE THREAD

**Issue by greg-szabo** [informalsystems/tendermint-rs#702]
We're relying on "latest" images of beta software for tendermint and we might also want gaia nodes in our CI testing. Best to have our own images so we supply the exact version we want to use.

**Comment by tony-iqlusion** [comment]
This could also be helpful with TMKMS, as generally we'd like to sync versions with tendermint-rs

**Comment by melekes** [comment]
> Best to have our own images so we supply the exact version we want to use.

Can't you already do that by using a specific tag: e.g. `tendermint/tendermint:v0.34.0` ?

**Comment by greg-szabo** [comment]
0.34.0 is not released yet (apparently it is now, it wasn't for a long time) and we had three different versions that we worked against. When we depended on `latest`, we ran into breaking changes that just landed on master, breaking our CI.

During stargate4 times, IBC depended on a commit that was not tagged in Tendermint Go, because the Stargate code used that specific commit.

With the latest stargate version, IBC depends on 0.34.0-rc5 (apparently in stargate-c and d, it's rc6) that doesn't have it's own docker image (Well, apparently rc6 does but I think rc5 might not.)

Last but not least, we might want to control the configuration on a more granular level, although that's questionable right now and it wasn't a good enough excuse so far.

So, although at this point most of these issues are resolved, the past few months we had a few pain points in CI because of inconsistent Docker images. I expect the same for the next release cycle.
